test/bigtable/family.js

```diff
@@ -0,0 +1,318 @@
+
+      assert.deepEqual(rule, {
+        maxAge: originalRule.age
+      });
+    });
+
+    it('should capture the max number of versions option', function() {
+      var originalRule = {
+        versions: 10
+      };
+
+      var rule = Family.formatRule_(originalRule);
+
+      assert.deepEqual(rule, {
+        maxNumVersions: originalRule.versions
+      });
+    });
+
+    it('should create a union rule', function() {
+      var originalRule = {
+        age: 10,
+        union: true
+      };
+
+      var rule = Family.formatRule_(originalRule);
+
+      assert.deepEqual(rule, {
+        union: {
+          rules: [{
+            maxAge: originalRule.age
+          }]
+        }
+      });
+    });
+
+    it('should create an intersecting rule', function() {
+      var originalRule = {
+        versions: 2,
+        intersection: true
+      };
+
+      var rule = Family.formatRule_(originalRule);
+
+      assert.deepEqual(rule, {
+        intersection: {
+          rules: [{
+            maxNumVersions: originalRule.versions
+          }]
+        }
+      });
+    });
+  });
+
+  describe('getMetadata', function() {
+    it('should return an error to the callback', function(done) {
+      var err = new Error('err');
+
+      family.parent.getFamilies = function(callback) {
+        callback(err);
+      };
+
+      family.getMetadata(function(err_) {
+        assert.strictEqual(err, err_);
+        done();
+      });
+    });
+
+    it('should update the metadata', function(done) {
+      var FAMILY = new Family(TABLE, FAMILY_NAME);
+      var response = {
+        families: {}
+      };
+
+      FAMILY.metadata = {
+        a: 'a',
+        b: 'b'
+      };
+
+      family.parent.getFamilies = function(callback) {
+        callback(null, [FAMILY], response);
+      };
+
+      family.getMetadata(function(err, metadata, apiResponse) {
+        assert.ifError(err);
+        assert.strictEqual(FAMILY.metadata, metadata);
+        assert.strictEqual(apiResponse, response);
+        done();
+      });
+    });
+
+    it('should throw a custom error', function(done) {
+      var response = {};
+
+      family.parent.getFamilies = function(callback) {
+        callback(null, [], response);
+      };
+
+      family.getMetadata(function(err, metadata, apiResponse) {
+        assert.strictEqual(err.code, 404);
+        assert.strictEqual(err.message,
+          'Column family not found: ' + family.id);
+        assert.strictEqual(metadata, null);
+        assert.strictEqual(apiResponse, response);
+        done();
+      });
+    });
+  });
+
+  describe('setMetadata', function() {
+    it('should provide the proper request options', function(done) {
+      family.request = function(protoOpts, reqOpts) {
+        assert.deepEqual(protoOpts, {
+          service: 'BigtableTableService',
+          method: 'updateColumnFamily'
+        });
+
+        assert.strictEqual(reqOpts.name, FAMILY_ID);
+        done();
```

Comment:
Test that the callback provided to `family.request` is the user's callback by executing it:

```js
family.request = function(protoOpts, reqOpts, callback) {
  ...
  callback(); // done()
};

family.setMetadata({}, done);
```